tests/test_cli.py

```diff
@@ -204,6 +204,20 @@ def test_cli_subscribe_logging(script_runner):
         assert "DEBUG" in ret.stderr
 
 
+def test_cli_interrupt(script_runner):
+    mock_consumer = MagicMock()
+    interrupt = MagicMock(side_effect=KeyboardInterrupt())
+    mock_consumer.__enter__ = interrupt
+    mock_consumer.__iter__ = interrupt
+    mock_consumer.__exit__ = interrupt
```

Comment:
Not too familiar with using side effects, but why is this done for all these methods rather than `__iter__` only?